### DISCUSSION TITLE
Correct term for built-in loader: `rspack:xxxx-loader` or `builtin:xxxx-loader`?

### DISCUSSION THREAD

**Discussion by hyf0** [web-infra-dev/rspack#2895] [Core] [answered]
Currently, Rspack is using `builtin:xxxx-loader` to represent the built-in loader.

- Would `builtin:xxx-loader` cause some misunderstanding?
- Is there a better pattern as a representation of the built-in loader?

@web-infra-dev/rspack cc

**Comment by hardfist** [comment]
or `native:xxx-loader`  shows it is written in rust

**Reply by hyf0** [reply]
or `naive:xxx-loader` shows we are too young too simple.

**Reply by hyf0** [reply]
Yeah. It sounds better than `rspack:xxx-loader` and emphasizes these plugins which are running in Rust.

**Comment by Boshen** [comment]
A few points: 
* `built-in` is an established industry word
* The end user doesn't care if it's written in Rust or JavaScript, as long as it is provided by the program, it's called a built-in
* Back in the old days, webpack provides built-in loaders such as `json-loader`, and they were referred as the built-in loader
* For developers (us), we can use the word `native-loader` to indicate it's written in Rust in our codebase / internal documentation

Please be careful with the terminology used for end users versus developers of the software.

**Reply by hyf0** [reply]
To be specific, the discussion are intend to decide how we represent the built-in loader in string in Rspack.

Currently, we are using [`builtin:xxx-loader`](https://github.com/web-infra-dev/rspack/pull/2894/files#diff-20fdab381355c53cd87d4b782507c4fd40b8c356a5deb5025f5eb9e8051ad79cR31).

and users would be affected by the conclusion of this discussion

```js
{
  loader: 'native:xxx-loader'
}
```

So, it's not just a term for communicating but also a JavaScript value that affects Rspack.

**Comment by hyf0** [comment, marked as answer]
We are going to stay with `builtin:xxx-loader`.
